src/samples/coap/lwm2m-client.c

```diff
@@ -281,23 +330,178 @@ read_security_server_obj(void *instance_data, void *user_data,
 }
 
 static int
+write_security_res(void *instance_data, void *user_data,
+    struct sol_lwm2m_client *client,
+    uint16_t instance_id, uint16_t res_id, const struct sol_lwm2m_resource *res)
+{
+    struct security_obj_instance_ctx *instance_ctx = instance_data;
+    int r = 0;
+
+    switch (res->id) {
+    case SECURITY_SERVER_SERVER_URI_RES_ID:
+        sol_blob_unref(instance_ctx->server_uri);
+        instance_ctx->server_uri = sol_blob_ref(res->data->blob);
+        break;
+    case SECURITY_SERVER_IS_BOOTSTRAP_RES_ID:
+        instance_ctx->is_bootstrap = res->data->b;
+        break;
+    case SECURITY_SERVER_SERVER_ID_RES_ID:
+        instance_ctx->server_id = res->data->integer;
+        break;
+    case SECURITY_SERVER_CLIENT_HOLD_OFF_TIME_RES_ID:
+        instance_ctx->client_hold_off_time = res->data->integer;
+        break;
+    case SECURITY_SERVER_BOOTSTRAP_SERVER_ACCOUNT_TIMEOUT_RES_ID:
+        instance_ctx->bootstrap_server_account_timeout = res->data->integer;
+        break;
+    default:
+        if (res->id >= 2 && res->id <= 9)
+            r = -ENOENT;
+        else
+            r = -EINVAL;
+    }
+
+    if (r >= 0)
+        printf("Resource written to Security object at /0/%" PRIu16 "/%" PRIu16 "\n",
+            instance_id, res->id);
+    return r;
+}
+
+static int
+write_security_tlv(void *instance_data, void *user_data,
+    struct sol_lwm2m_client *client,
+    uint16_t instance_id, struct sol_vector *tlvs)
+{
+    int r;
+    uint16_t i;
+    struct sol_lwm2m_tlv *tlv;
+    struct security_obj_instance_ctx *instance_ctx = instance_data;
+
+    SOL_VECTOR_FOREACH_IDX (tlvs, tlv, i) {
+        SOL_BUFFER_DECLARE_STATIC(buf, 64);
+
+        switch (tlv->id) {
+        case SECURITY_SERVER_SERVER_URI_RES_ID:
+            r = sol_lwm2m_tlv_get_bytes(tlv, &buf);
```

Comment:
Should I just check and abort with `SOL_INT_CHECK(r, < 0, -EINVAL);` for all this cases (Write TLV, Write Res and Create)?